backend/fms_core/viewsets/sample_lineage.py

```diff
@@ -0,0 +1,23 @@
+from tkinter import W
+from fms_core.services.sample_lineage import create_sample_lineage_graph
+from rest_framework import viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from django.core.exceptions import ValidationError
+
+from fms_core.models import Sample
+from fms_core.serializers import SampleSerializer, ProcessMeasurementSerializer
+
+class SampleLineageViewSet(viewsets.ViewSet):
+    @action(detail=True, methods=["get"])
+    def graph(self, _request, pk):
+        parent_sample = Sample.objects.get(pk=pk)
+        nodes, edges, errors = create_sample_lineage_graph(parent_sample)
+
+        if errors:
+            raise ValidationError(errors)
+        else:
+            return Response({
+                "nodes": [SampleSerializer(s).data for s in nodes],
```

Comment:
This is very onerous for the usage we make of it ... your get() will fetch many samples and processMeasurement without putting them in the store. The largest part of the information gathered will never be used. I would suggest to go one of two direction maybe @ckostiw and @sebastianballesteros can help see the best way. I would either 
1) return just the id (+ parent and child for edges), label (either sample name for nodes or protocol for edges) and maybe QC for some eye candy, but the idea is to carry as little as needed in a throwaway approach. 

2) go for the reuse approach and use redux action reducer to store your samples and processMeasurements in the store for reuse when processing the graph request. 

Also, you could also mix those 2 options and have a skeleton graph with ids and use the existing sample and processMeasurement action reducers to get their full representation in a single get().

Reply:
I tend to like the first option better. Given that there's no redux action/reducer at the moment, it might be better to just return the information needed (id, process measurements) to build the graph.